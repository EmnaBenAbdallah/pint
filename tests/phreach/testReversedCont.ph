
process a 1
process z 2

z 0 -> z 0 1
a 0 -> z 1 2

z 0 -> a 1 0
(*z 1 -> z 1 0*)


initial_state a 1, z 0



