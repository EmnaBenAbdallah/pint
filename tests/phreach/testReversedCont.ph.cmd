ph-reach -i testReversedCont.ph z 2
