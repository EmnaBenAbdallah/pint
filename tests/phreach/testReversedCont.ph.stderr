# testReversedCont.ph: 3 actions
# testing reachability of z_2 from state [a_1]
- computing aBS(z 0 2)... [ { a 0 } ]
- computing aBS(a 1 0)... [ { z 0 } ]
- computing aBS(z 0 0)... [ {  } ]
 - Sol(a 1 0) = [ { z 0 } ]
 - Sol(z 0 0) = [ {  } ]
 - Sol(z 0 2) = [ { a 0 } ]
 - Req(a 0) = [ a 1 0 ]
 - Req(z 0) = [ z 0 0 ]
 - Req(z 2) = [ z 0 2 ]
 - procs = { a 0, z 0, z 2 }
 - objs = { a 1 0, z 0 0, z 0 2 }

+ over-approximation (1) success
# handling concretion
- continuity saturation...
tBS: exact (expensive) computation.
- computing BS(z 0 2)... [ { z_0->z_0 1, a_0->z_1 2 } ]
reverse continuity from a 0: z 1 0
- new objectives [ z 1 0 ]
- computing aBS(z 1 0)... [  ]
 - Sol(a 1 0) = [ { z 0 } ]
 - Sol(z 0 0) = [ {  } ]
 - Sol(z 0 2) = [ { a 0 } ]
 - Req(a 0) = [ a 1 0 ]
 - Req(z 0) = [ z 1 0; z 0 0 ]
 - Req(z 2) = [ z 0 2 ]
 - Cont(a 1 0) = {  }
 - Cont(z 0 0) = {  }
 - Cont(z 0 2) = {  }
 - procs = { a 0, z 0, z 2 }
 - objs = { a 1 0, z 0 0, z 0 2 }

- new objectives [  ]
# continue concretizing
# aborting concretion
